include/sas.h

```diff
@@ -114,42 +110,46 @@ class SAS
   typedef uint64_t TrailId;
   typedef uint64_t Timestamp;
 
-#if HAVE_ZLIB_H
-  // Compression-related classes are only available if zlib is
+  // Compression profile
   class Profile
   {
   public:
-    inline Profile(std::string dictionary) : _dictionary(dictionary) {}
+    enum CompressionType {
+      ZLIB = 0,
+      LZ4
+    };
+
+    inline Profile(std::string dictionary, CompressionType t = ZLIB) : _dictionary(dictionary), _type(t) {}
+    inline Profile(CompressionType t) : _dictionary(""), _type(t) {}
     inline const std::string& get_dictionary() const {return _dictionary;}
+    inline bool is_lz4() const {return _type == LZ4;}
 
   private:
     const std::string _dictionary;
+    const CompressionType _type;
   };
 
   class Compressor
   {
   public:
-    static Compressor* get();
-
-    std::string compress(const std::string& s, const Profile* profile = NULL);
+    virtual std::string compress(const std::string& s, std::string dictionary) = 0;
+    static Compressor* get(const Profile* profile);
 
+  protected:
+    Compressor() {};
+    virtual ~Compressor() {};
+  
   private:
     static void init();
     static void destroy(void* compressor_ptr);
-
-    Compressor();
-    ~Compressor();
-
-    static const int WINDOW_BITS = 15;
-    static const int MEM_LEVEL = 9;
+    
+    static Compressor* get_zlib();
+    static Compressor* get_lz4();
```

Comment:
Shouldn't these methods live (as static methods) in LZ4Compressor and ZlibCompressor?